src/brasil/gov/agenda/tests/keywords.robot

```diff
@@ -54,6 +54,9 @@ Create AgendaDiaria
 
 Manage Portlets
     Go to   ${PLONE_URL}/@@manage-portlets
+    # Clica no botão para confirmar ação caso aparecer
+    ${present}=  Run Keyword And Return Status  Element Should Be Visible  name=form.button.confirm
+    Run Keyword If  ${present}  Click Button  form.button.confirm
```

Comment:
O estranho aqui é que um teste apresenta o botão, e o outro não.

Acredito que a diferença é que um está na raiz do site quando muda para ``@@manage-portlets`` e o outro não

Reply:
É muito difícil fazer debug da causa de porque isso ocorre no teste, nas não ocorre quando se roda `bin/instance` por exemplo. Chegamos a questionar o motivo desse diferenciação de comportamento quando é teste e quando não [há algum tempo](https://github.com/plone/plone.app.testing/issues/27) mas sem evolução.

Com relação aos falsos positivos, uma excelente leitura é http://devblog.4teamwork.ch/blog/2015/10/12/debugging-csrf-protection-false-positives-in-plone/

Reply:
Particularmente, o que tem de se analisar é: isso é um falso positivo? Não é?

Se não for, portanto pra mim o teste está ok;

Se não é, tem de entender porque dá o falso positivo e consertar. Obviamente, mundo de software não é perfeito: se vocês estão na correria pra gerar um release desse pacote (pois estamos aí no 2.x do IDG), eu abriria um relato para consertar esse "falso positivo" específico e colocaria um `FIXME` em cima dessa correção.

Reply:
ter certeza que isso aqui é utilizado para alguma coisa? eu acho que tem que ser removido.

Reply:
@hvelarde sim, temos 2 testes que quebram sem isso: https://travis-ci.org/plonegovbr/brasil.gov.agenda/builds/361350651?utm_source=github_status&utm_medium=notification

Reply:
mas então temos que consertar a causa do problema; esse pull request só muda o hack; prefiro deixar pulando o CSRF como estava.

Reply:
talvez precisamos desabilitar a checagem para estas views aqui https://github.com/plone/plone.app.portlets/blob/master/plone/app/portlets/browser/manage.py#L46

Reply:
> mas então temos que consertar a causa do problema; esse pull request só muda o hack; prefiro deixar pulando o CSRF como estava.

O que eu entendi é que o principal problema é que o hack funciona beleza aqui no github, mas aí quebra localmente... e a ficar setando variáveis de ambiente pra teste passar no ambiente local, aí também prefiro a mudança do `hack` como ele fez. Procede, @rodfersou ?

Reply:
@idgserpro esse é o código do ``@@manage-portlets``

Reply:
@idgserpro do jeito que está não quebra os testes localmente, eu estava testando errado quando disse que localmente funcionava antes.